pennylane/transforms/adjoint.py

```diff
@@ -113,20 +113,40 @@ def circuit():
 
     @wraps(fn)
     def wrapper(*args, **kwargs):
-        with get_active_tape().stop_recording(), QuantumTape() as tape:
-            fn(*args, **kwargs)
+        active_tape = get_active_tape()
+
+        if active_tape is not None:
+            with active_tape.stop_recording(), QuantumTape() as tape:
+                fn(*args, **kwargs)
+        else:
+            # Not within a queuing context
+            with QuantumTape() as tape:
+                fn(*args, **kwargs)
 
         if not tape.operations:
+            # we called op.expand(): get the outputted tape

```

Comment:
Basically if function outputs tape instead of queueing operations?

Reply:
Yes. Most commonly it's when `fn=op.expand` as called in the `except NotImplementedError` case when using recursion in `adjoint(op.expand)()`:
```python
            try:
                new_op = op.adjoint()
                adjoint_ops.append(new_op)
            except NotImplementedError:
                new_ops = adjoint(op.expand)()
```